fuel-merkle/src/common/position.rs

```diff
@@ -110,43 +114,84 @@ impl Position {
 
     /// Construct a position from a leaf index. The in-order index corresponding
     /// to the leaf index will always equal the leaf index multiplied by 2.
-    pub fn from_leaf_index(index: u64) -> Self {
-        Position(index * 2)
+    /// Panics if index is too large to fit into u64.
+    pub fn from_leaf_index(index: u64) -> Option<Self> {
+        Some(Position(index.checked_mul(2)?))
+    }
+
+    #[cfg(test)]
+    pub(crate) fn from_leaf_index_unwrap(index: u64) -> Self {
+        Self::from_leaf_index(index).expect("Index too large")
     }
 
     /// The sibling position.
     /// A position shares the same parent and height as its sibling.
-    pub fn sibling(self) -> Self {
-        let shift = 1 << (self.height() + 1);
-        let index = self.in_order_index() as i64 + shift * self.direction();
-        Self::from_in_order_index(index as u64)
+    pub fn sibling(self) -> Result<Self, GetNodeError> {
```

Comment:
You can put this function under `#[cfg(test)]`

Reply:
It is used in tests=)

Reply:
Done in 432e7c97